src/test/datascience/notebook/remoteNotebookEditor.vscode.common.ts

```diff
@@ -215,3 +163,69 @@ export function sharedRemoteNotebookEditorTests(
 
     return disposables;
 }
+
+export async function reopeningNotebookUsesSameRemoteKernel(
+    ipynbFile: Uri,
+    serviceContainer: IServiceContainer,
+    doNotSaveAndCloseNotebook = false
+) {
+    const remoteKernelIdProvider = serviceContainer.get<PreferredRemoteKernelIdProvider>(
+        PreferredRemoteKernelIdProvider
+    );
+
+    await openNotebook(ipynbFile);
+    await waitForKernelToGetAutoSelected(PYTHON_LANGUAGE, true);
+    let nbEditor = window.activeNotebookEditor!;
+    assert.isOk(nbEditor, 'No active notebook');
+    // Cell 1 = `a = "Hello World"`
+    // Cell 2 = `print(a)`
+    let cell2 = nbEditor.document.getCells()![1]!;
+    await Promise.all([
+        runAllCellsInActiveNotebook(),
+        waitForExecutionCompletedSuccessfully(cell2),
+        waitForTextOutput(cell2, 'Hello World', 0, false)
+    ]);
+
+    // Confirm kernel id gets saved for this notebook.
+    // This is not necessary, but this guarantees a faster & non-flaky test to ensure we don't close the notebook too early.
+    // If we nb it as soon as output appears, its possible the kernel id hasn't been saved yet & we mess that up.
+    // Optionally we could wait for 100ms.
+    await waitForCondition(
+        async () => !!remoteKernelIdProvider.getPreferredRemoteKernelId(nbEditor.document.uri),
+        5_000,
+        'Remote Kernel id not saved'
+    );
+
+    if (doNotSaveAndCloseNotebook) {
+        return;
+    }
+
```

Comment:
actually should just be two separate methods (one could call the other) since the part before this does not do what the name of the method says it does.